recipes/sqlitecpp/all/conanfile.py

```diff
@@ -41,9 +43,17 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
+        if self.options.has_codec:
+            self.options["sqlcipher"].enable_column_metadata = True
 
     def requirements(self):
-        self.requires("sqlite3/3.45.0")
+        if self.options.has_codec:
+            if Version(self.version) < "3.3.1":
+                raise ConanInvalidConfiguration("Using SQLCipher with this recipe is only available from version 3.3.1")
```

Comment:
This bit of code check should go into the `validate()` function instead of here, which allows proper discarding of invalid configurations :)